### PR TITLE
[FSTORE-1470] Documentation and Guides for Snowflake Schema

### PR DESCRIPTION
**Pull request by davitbzh**

### CODE REVIEW

File: docs/user_guides/fs/feature_view/query.md [hunk 1]
```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
...
+      feature vector = feature_view.get_feature_vector({
+        ‘cc_num’: “1234 5555 3333 8888”,
+        ‘issuer_id’: 20440455,
+        ‘merchant_id’: 44208484,
+        ‘account_id’: 84403331
+        })
+    ```
+
+#### Snowflake schema 
+Hopsworks also provides the possibility to define children and grandchildren of the root (the left most) feature group.
```
Comment (jimdowling) [inline docs/user_guides/fs/feature_view/query.md:154]: Hopsworks also provides the possibility to define a feature view that consists of a nested tree of children (to up to a depth of 10) from the root (left most) feature group.

Comment (jimdowling) [reply docs/user_guides/fs/feature_view/query.md:154]: Do we have a max depth for the tree?

File: docs/user_guides/fs/feature_view/query.md [hunk 2]
```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
...
+        ‘issuer_id’: 20440455,
+        ‘merchant_id’: 44208484,
+        ‘account_id’: 84403331
+        })
+    ```
+
+#### Snowflake schema 
+Hopsworks also provides the possibility to define children and grandchildren of the root (the left most) feature group. 
+This is called  Snowflake Schema data model where you need to build nested tables (subtrees) using joins, and then join the 
+subtrees to their parents which can be children of the root node (the leftmost feature group):
```
Comment (jimdowling) [inline docs/user_guides/fs/feature_view/query.md:156]: subtrees to their parents iteratively until you reach the root node (the leftmost feature group in the feature selection)

File: docs/user_guides/fs/feature_view/query.md [hunk 3]
```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
...
+
+=== "Python"
+    ```python
+        nested_selection = aggregated_cc_transactions.select_all()
+        .join(account_details.select_all())
+        .join(cc_issuer_details.select_all())
+
+        selected_features = credit_card_transactions.select_all()
+                .join(nested_selection)
+        .join(merchant_details.select_all()
```
Comment (jimdowling) [inline docs/user_guides/fs/feature_view/query.md:173]: Missing a closing parentheses "):

File: docs/user_guides/fs/feature_view/query.md [hunk 4]
```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
...
+    ```
+
+Now, you have the benefit that in online inference you only need to pass two foreign key values to retrieve the precomputed features:
```
Comment (jimdowling) [inline docs/user_guides/fs/feature_view/query.md:176]: pass two serving key values (the foreign keys of the leftmost feature group)

File: docs/user_guides/fs/feature_view/query.md [hunk 5]
```diff
@@ -114,48 +192,48 @@ For the Scala part of the API, equivalent methods are available in the `Feature`
...
 Filters are fully compatible with joins:
 
 === "Python"
     ```python
-    feature_join = rain_fg.select_all() \
-        .join(temperature_fg.select_all(), on=["date", "location_id"]) \
-        .join(location_fg.select_all(), left_on=["location_id"], right_on=["id"], join_type="left") \
-        .filter((rain_fg.location_id == 10) | (rain_fg.location_id == 20))
+    selected_features = credit_card_transactions_fg.select_all() \
+        .join(account_details_fg.select_all(), on=["cc_num"]) \
```
Comment (jimdowling) [inline docs/user_guides/fs/feature_view/query.md:208]: cc_num -> merchant_id

**Review by jimdowling — COMMENTED** [review]
Your data model is off.
Check here:
https://docs.google.com/document/d/12slqcD6Fu1DB47CTybAdCBO8NwocZGGK-wovm6mtKdk/edit

merchant_details should be joined on merchant_id
aggregated_cc_trans is a child of cc_transactions.
account_details is a child of aggregated_cc_trans

File: docs/user_guides/fs/feature_view/query.md [hunk 6]
```diff
@@ -75,35 +86,102 @@ By default, Hopsworks will use the maximal matching subset of the primary keys o
...
 The join key lists should contain the names of the features to join on.
 
 === "Python"
     ```python
-    feature_join = rain_fg.select_all() \
-        .join(temperature_fg.select_all(), on=["date", "location_id"]) \
-        .join(location_fg.select_all(), left_on=["location_id"], right_on=["id"], join_type="left")
+    selected_features = credit_card_transactions_fg.select_all() \
+        .join(account_details_fg.select_all(), on=["cc_num"]) \
+        .join(merchant_details_fg.select_all(), left_on=["cc_num"], right_on=["merchant_id"], join_type="left")
```
[the remark this answers is not in the document]
Comment (davitbzh) [reply docs/user_guides/fs/feature_view/query.md]: Regarding `join_type`, we are talking about more complex joins. So added comment about default join type and changed to `"inner"` just to make the point.

File: docs/user_guides/fs/feature_view/query.md (hunk 5 above)
Comment (davitbzh) [reply docs/user_guides/fs/feature_view/query.md:211]: its `account_details_fg`, cc_num is the foreign key here in `credit_card_transactions_fg` and `id` or `transaction_id` will be pk

**Review by rcnnnghm — APPROVED** [review]

**Review by rcnnnghm — APPROVED** [review]

**Review by jimdowling — APPROVED** [review]

**Pull request merged**